addons/website_livechat/models/mail_channel.py

```diff
@@ -30,20 +31,29 @@ def channel_info(self, extra_info=False):
         channel_infos = super(MailChannel, self).channel_info(extra_info)
         channel_infos_dict = dict((c['id'], c) for c in channel_infos)
         for channel in self:
-            visitor = channel.livechat_visitor_id
-            if visitor:
-                channel_infos_dict[channel.id]['visitor'] = {
-                    'name': visitor.display_name,
-                    'country_code': visitor.country_id.code.lower() if visitor.country_id else False,
-                    'country_id': visitor.country_id.id,
-                    'is_connected': visitor.is_connected,
-                    'history': self.sudo()._get_visitor_history(visitor),
-                    'website': visitor.website_id.name,
-                    'lang': visitor.lang_id.name,
-                    'partner_id': visitor.partner_id.id,
-                }
+            try:
+                visitor_info = channel._get_livechat_visitor_info()
+            except AccessError:
+                visitor_info = {}
+            if visitor_info:
+                channel_infos_dict[channel.id]['visitor'] = visitor_info
         return list(channel_infos_dict.values())
 
+    def _get_livechat_visitor_info(self):
```

Comment:
This is too close to `_channel_get_livechat_visitor_info` and should probably be refactored in master. I would be against introducing this new method here for this reason, and keep the code were it was and avoid unnecessary diff.

Reply:
If you want to avoid indentation level (which I think is a good idea too) and reduce diff to absolute minimum, what you can do is do "if not visitor: continue", then add the `try:` for which the rest of the code won't change indentation.

Reply:
Alternatively (and I think it's more usually done in python but I'm not sure), it's to pre-filter the loop variable so that the `if` is not needed at all.